### PR TITLE
Bevy 0.10

### PR DESCRIPTION
**Pull request by mvlabat**
Based on #148

### CODE REVIEW

**Comment by mvlabat** [comment]
I'm seeing the following warnings in the `ui` example:

> epaint: WARNING: pixels_per_point (dpi scale) have changed between text layout and tessellation. You must recreate your text shapes if pixels_per_point changes.

I'd be grateful if anyone could pick up this issue. I won't have time to return to the PR until the evening.

**Comment by DGriffin91** [comment]
I've been trying to track down the `pixels_per_point` thing. Haven't figured it out yet. This is the commit that added the warning:
https://github.com/emilk/egui/commit/39b63f6163c7aabbf20138b505f0c86d62977408

In the ui example, it comes from changing the `egui_settings.scale_factor` in `update_ui_scale_factor_system()`

We then set `egui_input.pixels_per_point` in `update_window_contexts()` which is called from `process_input_system()`

We get the shapes in `process_output_system()`
```rs
let full_output = ctx.end_frame();
let egui::FullOutput {
    platform_output,
    shapes,
    textures_delta,
    repaint_after,
} = full_output;

egui_render_output.shapes = shapes;
```

I think it only shows the warning for one frame if `scale_factor` changes.

File: examples/simple.rs [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+        // Systems that create Egui widgets should be run during the `CoreSet::Update` set,
+        // or after the `EguiSet::BeginFrame` system (which belongs to the `CoreSet::PreUpdate` set).
         .add_system(ui_example_system)
         .run();
 }
 
-fn ui_example_system(mut egui_context: ResMut<EguiContext>) {
-    egui::Window::new("Hello").show(egui_context.ctx_mut(), |ui| {
+fn ui_example_system(mut egui_ctx: Query<&mut EguiContext, With<PrimaryWindow>>) {
+    egui::Window::new("Hello").show(egui_ctx.single_mut().get_mut(), |ui| {
```
Comment (Vrixyz) [inline examples/simple.rs:15]: can it be simplified like this?

```suggestion
fn ui_example_system(egui_ctx: Query<&EguiContext, With<PrimaryWindow>>) {
    egui::Window::new("Hello").show(egui_ctx.single(), |ui| {
```

Comment (mvlabat) [reply examples/simple.rs:15]: Only if we implement `Deref` and `DerefMut` for `EguiContext`, giving people an interface to borrow the context immutably, which I'm hesitant to do due to the following:
https://github.com/mvlabat/bevy_egui/blob/7151c40f8564650fdfad5eaa15e72d055d0a5893/src/lib.rs#L247-L256

**Comment by mvlabat** [comment]
@DGriffin91 thanks for looking into this! That definitely saved me time investigating. It seems that I was able to fix the warning here: 7151c40.

**Comment by mvlabat** [comment]
The PR is almost finished. I'll let people review the API changes, and tomorrow morning I'll push the last bits of the documentation fixes and make a release.

File: examples/simple.rs (hunk 1 above)
Comment (mvlabat) [reply examples/simple.rs]: I ended up introducing the `EguiContexts` system param, to combine the best of both worlds: low boilerplate and mutable borrowing by default.

**Pull request merged**